blockchain/chain/src/store/chain_store.rs

```diff
@@ -423,6 +434,80 @@ where
     })
 }
 
+pub fn get_chain_message<DB>(db: &DB, key: &Cid) -> Result<Box<dyn Message>, Error>
+where
+    DB: BlockStore,
+{
+    let value = db.read(key.key())?;
+    let bytes = value.ok_or_else(|| Error::UndefinedKey(key.to_string()))?;
+    let unsigned_message: Result<UnsignedMessage, _> = from_slice(&bytes);
+    if let Ok(s) = unsigned_message {
+        Ok(Box::new(s))
+    } else {
+        let signed_message: SignedMessage = from_slice(&bytes)?;
+        Ok(Box::new(signed_message))
+    }
+}
+
+pub fn message_for_tipset<DB>(db: &DB, ts: &Tipset) -> Result<Vec<Box<dyn Message>>, Error>
+where
+    DB: BlockStore,
+{
+    let mut applied: HashMap<Address, u64> = HashMap::new();
+    let mut balances: HashMap<Address, BigUint> = HashMap::new();
+    let state = StateTree::new_from_root(db, ts.parent_state())?;
+
+    //message to get all messages for block_header into a single iterator
+    type BoxMessage = Box<dyn Message>;
```

Comment:
This type is only used in the line below, consider removing?

Reply:
It should be moved up, used in other functions as well